tests/core/mountpoint/test_base.py

```diff
@@ -523,3 +523,40 @@ def test_nested_rw_access(mountpoint_service):
         with open(str(fpath), "rb") as fin:
             data = fin.read()
             assert data == b"whatever"
+
+
+@pytest.mark.trio
+@pytest.mark.slow
+@pytest.mark.mountpoint
+@pytest.mark.parametrize("n", [10, 100, 1000])
+@pytest.mark.parametrize("base_path", ["/", "/foo"])
+async def test_mountpoint_iterdir_with_many_files(
+    n, base_path, base_mountpoint, alice_user_fs, event_bus
+):
+    wid = await alice_user_fs.workspace_create("w")
+    workspace = alice_user_fs.get_workspace(wid)
+    await workspace.mkdir(base_path, parents=True, exist_ok=True)
+    names = [f"some_file_{i:03d}.txt" for i in range(n)]
```

Comment:
What's the need of having multiple values for `n` ?

Reply:
Typically, the problem fixed in this PR appears with more than ~100 files in a non-root directory. I felt the best way to cover those weird cases is to parametrize. The parameters can be adjusted over time.

Reply:
👍